### PR TITLE
[th/eval-result-rework] cleanup evaluation of results

### PR DESCRIPTION
**Pull request by thom311**
- extract parsing of iperf and netperf output to (unit tested) helper functions
- let the parsing don't fail with an exception, instead signal parsing errors in the result (`success=False`).
- let the test already compute the `Bitrate` and embed it in the result. This moves the functionality from the `Evaluator` right to the place where we collect the test result. It also means, the result JSON contains also the bit rate.
- implement bitrate and evaluation for `netperf` test type.

### CODE REVIEW

File: common.py [hunk 1]
```diff
@@ -557,3 +557,49 @@ def ethtool_stat_get_packets(data: dict[str, str], packet_type: str) -> Optional
     if not any_match:
         return None
     return total_packets
+
+
+def netperf_parse(testname: Any, data: str) -> dict[str, float]:
+
+    # We want to accept both the testname as string or as TestType.  Note that
```
Comment (wizhaoredhat) [inline common.py:564]: Why does testname have 2 possible types?

File: tests/test_common.py [hunk 2]
```diff
@@ -492,3 +492,72 @@ def test_ethtool_parse_stat() -> None:
...
+    assert common.netperf_parse("TCP_STREAM", data) == {
+        "Receive Socket Size Bytes": 131072.0,
+        "Send Socket Size Bytes": 16384.0,
+        "Send Message Size Bytes": 16384.0,
+        "Elapsed Time Seconds": 30.0,
+        "Throughput 10^6bits/sec": 35710.68,
+    }
+
+    with pytest.raises(TypeError):
+        common.netperf_parse("xxx", data)
```
Comment (wizhaoredhat) [inline tests/test_common.py:552]: What is "xxx"?

File: common.py (hunk 1 above)
Comment (thom311) [reply common.py:564]: this is in `common.py`, but `TestType` is in `tftbase.py`. `common.py` cannot use `tftbase.py`.

For the most part, `testname` is just a string and that's it.

However, the callers actually have a `TestType.NETPERF_*` enum at hand. Allow them to pass that enum directly. `common.netperf_parse()` doesn't really know about that enum, but it doesn't care. It only cares that `str(testname)` is one of the expected arguments.

File: tests/test_common.py (hunk 2 above)
Comment (thom311) [reply tests/test_common.py:552]: an invalid argument `testname` for the unit test, that results in a `TypeError` exception.

File: common.py (hunk 1 above)
Comment (thom311) [reply common.py:564]: Alternatives:

- move the code to `tftbase.py` and have `testname: str | TestType` (or only `testname: TestType`). I didn't do that, because this code is for the most part independent from ocp-t-f-t (hence in `common.py`). Also, `TestType` contains other values like `HTTP`, so accepting `TestType` is not great.

- only requiest a string. Then the caller must somehow convert the `TestType` that they have to the right string.

The current alternative is chosen, because it's convenient for the caller (they can just pass their `TestType` enum), while the parsing of the input `testname` is still quite trivial (convert to string, and compare that in a relaxed way).

File: tests/test_common.py (hunk 2 above)
Comment (thom311) [reply tests/test_common.py]: changed `"xxx"` to `"bogus-name"`

File: common.py (hunk 1 above)
Comment (wizhaoredhat) [reply common.py:564]: It is kind of unexpected to see this code in common.py since it is specific for netperf. Could we move the code to related netperf code?

Comment (thom311) [reply common.py:564]: The only place that has netperf related code is `testTypeNetPerf.py`. That module depends on other ocp-t-f-t code. `testTypeNetPerf.py` is not simple, because you need to understand how it integrates into ocp-t-f-t.

Moving code away from `testTypeNetPerf.py` makes that `testTypeNetPerf.py` a bit simpler -- which is the most important aspect, because `testTypeNetPerf.py` is the complicated part. It also encourages to make the code self-contained, unit tested and handle edge cases. Which it could have been also while residing inside `testTypeNetPerf.py`, but it wasn't (point in case!).

There is also `common.ethtool_stat_parse()`. If we dislike this, then we should add `ethtool_utils.py`, `netperf_utils.py` (or leave them in `pluginValidateOffload.py` and `testTypeNetPerf.py`?). I don't see value in splitting up `common.py` (or keeping functionality out of it), as long as all the pieces are self-contained, Python-stdlib-only, simple components, and do a clear thing. Unlike most other modules, `common.py`/`tftbase.py` does not require that it's functions share a common topic/purpose.

Comment (thom311) [reply common.py]: Despite what I said. The code now moved away from `common.py` and branch rebased. Better?

**Comment by wizhaoredhat** [comment]
LGTM, thanks @thom311 !

File: common.py (hunk 1 above)
Comment (wizhaoredhat) [reply common.py]: Looks good!
